src/ensembl/production/datacheck/exceptions.py

```diff
@@ -0,0 +1,20 @@
+# See the NOTICE file distributed with this work for additional information
+#   regarding copyright ownership.
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+from os import getenv
+from requests.exceptions import HTTPError
+
+
+class MissingIndexException(HTTPError):
+
+    def __init__(self, *args, **kwargs) -> None:
+        kwargs['message'] = f"Missing Datacheck index configuration for {getenv('ENS_VERSION')}"
+        super().__init__(*args, **kwargs)
```

Comment:
This triggers a weird error when raised: `TypeError: MissingIndexException() takes no keyword arguments`.
It's because of how `HTTPError` is [defined](https://2.python-requests.org/en/master/_modules/requests/exceptions/#RequestException), it's sligtly misleading but built-in Python Exceptions don't take keyword arguments so neither their derived classes should. In this case not sure what the keyword `'message'` comes from.

It could be fixed with:
```python
    # Note: kwargs can only have 'request' and/or 'response' keywords when invoked
    def __init__(self, *args, **kwargs) -> None:
        super().__init__(f"Missing Datacheck index configuration for {getenv('ENS_VERSION')}", **kwargs)
```
But this would make it unusable for raising custom messages when invoked.

So a better (and more idiomatic) solution would be defining a custom exception and format the message in a dedicated `errorhandler`.

```suggestion

class MissingIndexException(IOError):
    pass
```